### PR TITLE
chore: fix build badge

### PR DESCRIPTION
**Pull request by mlavacca**
**What this PR does / why we need it**:

**Which issue this PR fixes**:

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

**Review by pmalek — DISMISSED** [review]
We could consider renaming the workflow since `PR checks` doesn't really fit in here when displayed on project's README 🤔

**Comment by czeslavo** [comment]
> We could consider renaming the workflow since `PR checks` doesn't really fit in here when displayed on project's README 🤔

👍 "PR checks" name is too specific for this workflow as we run it not only for PRs.

All the jobs that we run in it are:
```yaml
      - up-to-date
      - tools
      - linters
      - unit-tests
      - envtest-tests
      - integration-tests
      - conformance-tests
      - build-docker-image
```
Based on that I think we could rename it to just "Checks", "Verification", or something similar.

**Comment by mlavacca** [comment]
> We could consider renaming the workflow since PR checks doesn't really fit in here when displayed on project's README 🤔

Yep, I agree. As @czeslavo suggested, I renamed it just "Checks".

File: .github/workflows/pr_checks.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-name: PR checks
-run-name: PR checks, branch:${{ github.ref_name }}, triggered by @${{ github.actor }}
+name: Checks
```
Comment (czeslavo) [inline .github/workflows/pr_checks.yaml:1]: Shouldn't we also rename the file itself?

File: README.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
-[![][kong-logo]][kong-url]
-[![Build Status](https://github.com/kong/kubernetes-ingress-controller/workflows/Test/badge.svg)](https://github.com/kong/kubernetes-ingress-controller/actions?query=branch%3Amain+event%3Apush)
+[![kong-logo]][kong-url]
+[![Build Status](https://github.com/Kong/kubernetes-ingress-controller/actions/workflows/pr_checks.yaml/badge.svg)](https://github.com/Kong/kubernetes-ingress-controller/actions/workflows/pr_checks.yaml)
```
Comment (czeslavo) [inline README.md:2]:
```suggestion
[![Build Status](https://github.com/Kong/kubernetes-ingress-controller/actions/workflows/checks.yaml/badge.svg)](https://github.com/Kong/kubernetes-ingress-controller/actions/workflows/checks.yaml)
```

**Review by czeslavo — APPROVED** [review]

**Pull request merged**